tinygrad/nn/state.py

```diff
@@ -220,17 +223,17 @@ def find_class(self, module, name):
       return intercept[name] if module_root == "torch" else super().find_class(module, name)
     def persistent_load(self, pid): return deserialized_objects.get(pid, pid)
 
-  if zipfile.is_zipfile(fn):
-    myzip = zipfile.ZipFile(fn, 'r')
+  if passthrough_reset(zipfile.is_zipfile(fobj)):
+    myzip = zipfile.ZipFile(fobj, 'r')
     base_name = myzip.namelist()[0].split('/', 1)[0]
     for n in myzip.namelist():
       if n.startswith(f'{base_name}/data/'):
         with myzip.open(n) as myfile:
           offsets[n.split("/")[-1]] = myfile._orig_compress_start # type: ignore
     with myzip.open(f'{base_name}/data.pkl') as myfile:
       return TorchPickle(myfile).load()
-  elif tarfile.is_tarfile(fn):
-    with tarfile.open(fn, "r") as tar:
+  elif passthrough_reset(tarfile.is_tarfile(fobj)):
```

Comment:
Err, do you want this reset to be on the open?

Reply:
`passthrough_reset` is not required `tarfile.is_tarfile`, it resets correctly.

I made a mistake when testing this, I had failing tests without `passthrough_reset` `tarfile.is_tarfile` for some reason.